ensembleperturbation/perturbation/atcf.py

```diff
@@ -1637,30 +1905,36 @@ def write_perturbed_track(
                         f'perturbation type "{variable.perturbation_type}" is not recognized'
                     )
 
-                if isinstance(variable, MaximumSustainedWindSpeed):
-                    # In case of Vmax need to change the central pressure in accordance with Holland B relationship
-                    dataframe[CentralPressure.name] = self.compute_pc_from_Vmax(dataframe)
-                elif isinstance(variable, RadiusOfMaximumWinds):
-                    # In case of Rmax need to change the r34/50,64 radii at all quadrants in the same way
-                    quadrants = [
-                        IsotachRadiusNEQ(),
-                        IsotachRadiusSEQ(),
-                        IsotachRadiusSWQ(),
-                        IsotachRadiusNWQ(),
-                    ]
-                    for quadrant in quadrants:
-                        dataframe = quadrant.perturb(
-                            vortex_dataframe=dataframe,
-                            values=perturbed_values,
-                            times=self.validation_times,
-                            inplace=True,
-                        )
-
         # remove units from data frame
         for column in dataframe:
             if isinstance(dataframe[column].dtype, PintType):
                 dataframe[column] = dataframe[column].pint.magnitude
 
+        # enter if it is not the original unperturbed file
+        if any(~numpy.isnan(list(perturbation.values()))):
+            # Compute potential changes in the central pressure in accordance with Holland B relationship
+            dataframe[CentralPressure.name] = self.compute_pc_from_Vmax(dataframe)
```

Comment:
Moved the central pressure change in accordance with Vmax change outside of variable perturbation loop where it was called after Vmax was perturbated. It is possible the small change in trajectory can affect the radial Vmax and hence the central pressure, so this change will mean the function call is not dependent on order of perturbation (Vmax before or after along_track and cross_track perturbations).